modules/graphics/weston.nix

```diff
@@ -9,5 +9,112 @@
   environment.noXlibs = false;
   environment.systemPackages = with pkgs; [
     weston
+    # Seatd is needed to manage log-in process for weston
+    seatd
   ];
+
+  # Next 4 services/targets are taken from official weston documentation:
+  # https://wayland.pages.freedesktop.org/weston/toc/running-weston.html
+  #
+  # To run weston, after log-in to VT or SSH run:
+  # systemctl --user start weston.service
+  #
+  # I am pretty sure it is possible to have it running automatically, I just
+  # haven't found the way yet.
+
+  # Weston socket
+  systemd.user.sockets."weston" = {
+    unitConfig = {
+      Description = "Weston, a Wayland compositor";
+      Documentation = "man:weston(1) man:weston.ini(5)";
+    };
+    socketConfig = {
+      ListenStream = "%t/wayland-0";
+    };
+    wantedBy = ["weston.service"];
+  };
+
+  # Weston service
+  systemd.user.services."weston" = {
+    enable = true;
+    description = "Weston, a Wayland compositor, as a user service TEST";
+    documentation = ["man:weston(1) man:weston.ini(5)" "https://wayland.freedesktop.org/"];
+    requires = ["weston.socket"];
+    after = ["weston.socket" "ghaf-session.service"];
+    serviceConfig = {
+      # Previously there was "notify" type, but for some reason
+      # systemd kills weston.service because of timeout (even if it is disabled).
+      # "simple" works pretty well, so let's leave it.
+      Type = "simple";
+      #TimeoutStartSec = "60";
+      #WatchdogSec = "20";
+      # Defaults to journal
+      StandardOutput = "journal";
+      StandardError = "journal";
+      ExecStart = "${pkgs.weston}/bin/weston";
+    };
+  };
+
+  systemd.user.targets."ghaf-session" = {
+    description = "Ghaf graphical session";
+    bindsTo = ["ghaf-session.target"];
+    before = ["ghaf-session.target"];
+  };
+
+  systemd.services."ghaf-session" = {
+    description = "Ghaf graphical session";
+
+    # Make sure we are started after logins are permitted.
+    after = ["systemd-user-sessions.service"];
+
+    # if you want you can make it part of the graphical session
+    #Before=graphical.target
+
+    # not necessary but just in case
+    #ConditionPathExists=/dev/tty7
+
+    serviceConfig = {
+      Type = "simple";
+      Environment = "XDG_SESSION_TYPE=wayland";
+      ExecStart = "${pkgs.systemd}/bin/systemctl --wait --user start ghaf-session.target";
+
+      # The user to run the session as. Pick one!
+      User = "ghaf";
```

Comment:
https://github.com/tiiuae/ghaf/blob/5b8868b3a5741efeee3b9239306fce19d4a8429d/modules/development/authentication.nix#L6-L7

Reply:
Probably not, since it is hidden in the `let ... in` scope.
In theory, the whole process of running `weston` should be different, but since we do not have other normal users except `ghaf`, I think it is OK to use it as a temporary solution. In future we either run `weston` from VM (where it does not matter which user is used) or even use other DM than `weston`.

Reply:
I like your thinking here. In general we should create an example of running weston as different weston-specific normal user, not `ghaf` but for now, I digress.

Reply:
Please note release-build doesn't currently have ghaf-user